### PR TITLE
Update lgogdownloader to 3.15

### PR DESCRIPTION
**Pull request by Peppersawce**
This new recipe, along with the fixes in #11203 and #11204, solves all issues that are present witht the current package (bug #11201)

Currently the recipe builds the command-line utility just fine but the Qt Gui now wants Qt6. I haven't touched that part of the recipe at all.

### CODE REVIEW

File: games-util/lgogdownloader/patches/lgogdownloader-3.12.patchset [hunk 1]
```diff
@@ -1,77 +0,0 @@
-From c65d24ff976fc9914b2b92bb31e0e07844180aaf Mon Sep 17 00:00:00 2001
-From: Begasus <begasus@gmail.com>
-Date: Sat, 21 Oct 2023 21:30:55 +0200
-Subject: Fix for: help2man: can't get `--help' info (for manpage)
```
Comment (Begasus) [inline games-util/lgogdownloader/patches/lgogdownloader-3.12.patchset:4]: Not needed anymore?

File: games-util/lgogdownloader/patches/lgogdownloader-3.15.patchset [hunk 2]
```diff
@@ -0,0 +1,54 @@
+From 357d5eaabc6ec3f1d8af554eacba6e30bc35f924 Mon Sep 17 00:00:00 2001
+From: Peppersawce <michaelpeppers89@yahoo.it>
+Date: Wed, 23 Oct 2024 12:11:13 +0200
+Subject: Applying Mika T. Lindqvist's Iconv fix to 3.15
```
Comment (Begasus) [inline games-util/lgogdownloader/patches/lgogdownloader-3.15.patchset:4]: Keep credits to previous committer.

File: games-util/lgogdownloader/patches/lgogdownloader-3.12.patchset (hunk 1 above)
Comment (Peppersawce) [reply games-util/lgogdownloader/patches/lgogdownloader-3.12.patchset:4]: Yeah, the code in that section is totally different now and the man page works without patching :)

Comment (Begasus) [reply games-util/lgogdownloader/patches/lgogdownloader-3.12.patchset:4]: Probably fixed upstream then, thanks!

File: games-util/lgogdownloader/patches/lgogdownloader-3.15.patchset (hunk 2 above)
Comment (Peppersawce) [reply games-util/lgogdownloader/patches/lgogdownloader-3.15.patchset]: Done :)

**Comment by Peppersawce** [comment]
Welp, I found out this Pull Request is a duplicate of #11247, so I'm closing this one

**Pull request closed without merging**